CoursesManager.UI/Service/ConfigurationService.cs

```diff
@@ -0,0 +1,184 @@
+
+        }
+
+
+        public string SafeDecrypt(string encryptedText)
+        {
+            try
+            {
+                return _encryptionService.Decrypt(encryptedText);
+            }
+            catch
+            {
+                Console.WriteLine("Fout bij decryptie. Mogelijk onjuiste sleutel of waarde niet versleuteld.");
+                return encryptedText; 
+            }
+        }
+
+
+        public EnvModel GetDecryptedEnvSettings()
+        {
+            EnvManager<EnvModel>.Values.ConnectionString =
+                SafeDecrypt(EnvManager<EnvModel>.Values.ConnectionString);
+
+            EnvManager<EnvModel>.Values.MailConnectionString =
+                SafeDecrypt(EnvManager<EnvModel>.Values.MailConnectionString);
+
+            return EnvManager<EnvModel>.Values;
+        }
+
+
+
+
+
+        public Dictionary<string, string> GetDatabaseParameters()
+        {
+            var decryptedConfig = GetDecryptedEnvSettings();
+            if (string.IsNullOrWhiteSpace(decryptedConfig.ConnectionString))
+            {
+                Console.WriteLine("Geen ConnectionString gevonden in de gedecrypteerde instellingen.");
+                return new Dictionary<string, string>();
+            }
+
+            return ParseConnectionString(decryptedConfig.ConnectionString);
+        }
+
+        public Dictionary<string, string> GetMailParameters()
+        {
+            var decryptedConfig = GetDecryptedEnvSettings();
+            if (string.IsNullOrWhiteSpace(decryptedConfig.MailConnectionString))
+            {
+                Console.WriteLine("Geen MailConnectionString gevonden in de gedecrypteerde instellingen.");
+                return new Dictionary<string, string>();
+            }
+
+            return ParseConnectionString(decryptedConfig.MailConnectionString);
+        }
+
+
+
+        public bool ValidateSettings()
+        {
+            
+            var envFilePath = Path.Combine(Directory.GetCurrentDirectory(), ".env");
+            if (!File.Exists(envFilePath))
+            {
+                Console.WriteLine("Geen .env-bestand gevonden. Gebruiker moet instellingen invullen.");
+                return false; 
+            }
+
+            
+            try
+            {
+                var config = GetDecryptedEnvSettings();
+
+                
+                if (string.IsNullOrWhiteSpace(config.ConnectionString) ||
+                    string.IsNullOrWhiteSpace(config.MailConnectionString))
+                {
+                    Console.WriteLine("De ConnectionString of MailConnectionString is leeg of ongeldig.");
+                    return false; 
+                }
+
+                
+                if (!ValidateConnectionString(config.ConnectionString))
+                {
+                    Console.WriteLine("De ConnectionString is ongeldig.");
+                    return false;
+                }
```

Comment:
Hier moet je eigenlijk ook nog even var _ = new MySQLConnection(connectionstring) doen. Als dat het password of wat dan ook fout is krijg je een exception